addons/edi/models/edi_record.py

```diff
@@ -187,7 +199,13 @@ def _add_edi_relates(self, required=True):
                 for key, recs in batch.groupby(keygetter):
                     target = targets_by_key.get(key)
                     if required and not target:
-                        target = recs.missing_edi_relates(rel, key)
+                        try:
+                            target = recs.missing_edi_relates(rel, key)
+                        except UserError as ex:
+                            if doc.fail_fast:
+                                raise
+                            _logger.warning('Suppressed error for missing relate %r, %r, %r', recs, rel, key)
```

Comment:
More single quotes, might be worth black formatting the module again.

Reply:
Reformatted entirely both edi and edi_sale now.